src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/profile/Profile.java

```diff
@@ -0,0 +1,387 @@
+      case NB_BATHROOMS:
+        return nbBathroomsWeightRange;
+      case NB_BEDROOMS:
+        return nbBedroomsWeightRange;
+      case NB_SLEEPING:
+        return nbSleepingWeightRange;
+      case NB_MIN_NIGHT:
+        return nbMinNightSubjectiveValueWeightRange;
+      case PRICE_PER_NIGHT:
+        return pricePerNightSubjectiveValueWeightRange;
+      default:
+        throw new IllegalArgumentException();
+    }
+  }
+
+  /**
+   * Get the middle of the given Range
+   *
+   * @param range the Range
+   * @return the middle of the given Range of Doubles
+   */
+  static double getMiddleOfRange(Range<Double> range) {
+    checkNotNull(range);
+    return (range.upperEndpoint() + range.lowerEndpoint()) / 2;
+  }
+
+  /**
+   * Get the middle of the range of the given criterion
+   *
+   * @param crit the Criterion
+   * @return the middle of the Range of the given Criterion
+   */
+  double getMiddleOfRange(Criterion crit) {
+    return getMiddleOfRange(this.getWeightRange(crit));
+  }
+
+  /**
+   * Sets the linearAVF of a Profile
+   *
+   * @return Profile with its LinearAVF set
+   */
+  public Profile setLinearAVF(LinearAVF newLinearAvf) {
+    Profile prof = cloneProfile();
+
+    prof.linearAvf = newLinearAvf;
+
+    return prof;
+  }
+
+  /**
+   * Sets the subjective value weight of a criterion
+   *
+   * @param crit the criterion we want to know the value
+   * @return the subjective value weight
+   */
+  public Profile setWeightRange(Criterion crit, Range<Double> value) {
+    checkRangeValidity(value);
+    Profile prof = cloneProfile();
+    switch (crit) {
+      case TELE:
+        prof.setTeleSubjectiveValueWeight(value);
+        break;
+      case TERRACE:
+        prof.setTerraceSubjectiveValueWeight(value);
+        break;
+      case WIFI:
+        prof.setWifiSubjectiveValueWeight(value);
+        break;
+      case FLOOR_AREA:
+        prof.setFloorAreaSubjectiveValueWeight(value);
+        break;
+      case FLOOR_AREA_TERRACE:
+        prof.setFloorAreaTerraceSubjectiveValueWeight(value);
+        break;
+      case NB_BATHROOMS:
+        prof.setNbBathroomsSubjectiveValueWeight(value);
+        break;
+      case NB_BEDROOMS:
+        prof.setNbBedroomsSubjectiveValueWeight(value);
+        break;
+      case NB_SLEEPING:
+        prof.setNbSleepingSubjectiveValueWeight(value);
+        break;
+      case NB_MIN_NIGHT:
+        prof.setNbMinNightSubjectiveValueWeight(value);
+        break;
+      case PRICE_PER_NIGHT:
+        prof.setPricePerNightSubjectiveValueWeight(value);
+        break;
+      default:
+        throw new IllegalArgumentException();
+    }
+
+    return prof;
+  }
+
+  /**
+   * check if the given range is valid for this context
+   *
+   * @param value the Range to check
+   * @throws IllegalArgumentException when doubles contained are < 0 or if there are only 1 value in
+   *     the range
+   */
+  private void checkRangeValidity(Range<Double> value) {
```

Comment:
Il faut s’assurer également que la valeur du lower endpoint est finie.

Reply:
Il me semble que cette vérification est déjà effectué par la ligne suivante : 
checkArgument(value.hasLowerBound() && value.hasUpperBound(), "The given range is not valid");

Reply:
Je ne pense pas. La valeur lower bound peut être présente et infinie négative, par exemple. À vérifier.